### PR TITLE
Spring Boot aplication class name is now based on app name.

### LINKED ISSUE #2868: Application class should be named according to app name
With micro services applications being introduced in JHipster 3.0, it will be quite common to have several JHipster applications running at same time in your IDE in its own tab, as each tab is named according to the main class it means that all tabs will be named Application.

I propose to rename Application.java according to app base name.

At least, it should be done for Gateway.

I can do it if agreed.

### PR DESCRIPTION
**Pull request by gmarziou**
This makes easier to run several applications in IDE by having tabs named differently rather than having all of them named 'Application'.

Tested with maven and gradle.
Generated a monolith app and a gateway.

Close #2868

### CODE REVIEW

File: generators/server/index.js [hunk 1]
```diff
@@ -522,6 +522,11 @@ module.exports = JhipsterServerGenerator.extend({
             this.camelizedBaseName = _.camelize(this.baseName);
             this.slugifiedBaseName = _.slugify(this.baseName);
             this.lowercaseBaseName = this.baseName.toLowerCase();
+            
+            this.mainClass = _.capitalize(this.camelizedBaseName);
+            if (!this.camelizedBaseName.endsWith('App')) {
+               this.mainClass += 'App';
```
Comment (deepu105) [inline generators/server/index.js]: We need to do this check for angular app name ad well, currently we always append App to basename for that

**Comment by deepu105** [comment]
`npm test` failing

```
92 passing (15s)
  14 failing
  1) JHipster generator default configuration creates expected default files:
      AssertionError: src/main/java/com/mycompany/myapp/Application.java, no such file or directory
      + expected - actual
      -false
      +true
```

File: generators/server/index.js (hunk 1 above)
Comment (deepu105) [inline generators/server/index.js]: I will add this check there

**Comment by gmarziou** [comment]
Forgot 'npm test', I'll look at it tonight.

**Pull request merged**